promql/lex.go

```diff
@@ -551,6 +561,44 @@ func lexInsideBraces(l *lexer) stateFn {
 	return lexInsideBraces
 }
 
+// lexValueSequence scans a value sequence of a series description.
+func lexValueSequence(l *lexer) stateFn {
+	switch r := l.next(); {
+	case r == eof:
+		return lexStatements
```

Comment:
I guess `lexStatements` will just get `eof` again, so you're letting that handle it correctly instead of emitting `itemEOF` here and stopping the parsing?